tmt/utils.py

```diff
@@ -870,7 +871,7 @@ def listify(
     if isinstance(data, dict):
         for key in keys or data:
             if key in data:
-                data[key] = listify(data[key], split=split)
+                data[key] = listify(data[key], split=split)  # type: ignore
```

Comment:
I have 2 notes regarding this function.

First, it doesn't make much sense to have return type `StepData` alongside `Dict` and `List` due to the character of the function (how it transforms the input data). If you feed it `StepData`, it will change it into something that is not `StepData` type anymore:

E.g. code `print(tmt.utils.listify(tmt.steps.StepData(name='hello', order=None)))` prints out: `{'name': ['hello'], 'order': [None]}` which is clearly not `StepData` anymore due to the nested lists. The only way to get `StepData` as an output seems to be by inputting an empty dictionary but why would one use this function then? :)

Second, if we convert `TypedDict` to `Dict`, we can get rid of the mypy complaint that made us insert the `type: ignore`. Converting to `Dict` IMO makes sense due to 1).

My proposal is to change the function to this:
```
def listify(
        data: Union[Tuple[Any, ...], List[Any], str, Dict[Any, Any], 'tmt.steps.StepData'],
        split: bool = False,
        keys: Optional[List[str]] = None) -> Union[List[Any], Dict[Any, Any]]:
    """
    Ensure that variable is a list, convert if necessary

    For dictionaries check all items or only those with provided keys.
    Also split strings on white-space/comma if split=True.
    """
    separator = re.compile(r'[\s,]+')
    if isinstance(data, tuple):
        data = list(data)
    if isinstance(data, list):
        return fmf.utils.split(data, separator) if split else data
    if isinstance(data, str):
        return fmf.utils.split(data, separator) if split else [data]
    if isinstance(data, dict):
        data_dict = dict(data)
        for key in keys or data_dict:
            if key in data_dict:
                data_dict[key] = listify(data_dict[key], split=split)
        return data_dict
    return [data]
```

Reply:
I totally get your point about why providing `StepData` as another accepted type for `data` in `listify(...)` may not make much sense, and I agree with it. But since we discussed this PR in the team gathering with @happz and @thrix and agreed on adding 'tmt.steps.StepData' as another accepted type for `data` in `listify`, I'd like to hear their thoughts before I make further changes.

Reply:
It makes sense to **accept** `StepData`, we already discussed it on team meetings I believe. But it doesn't make much sense to **return** `StepData`. See how it transforms and outputs the data - they do not resemble `StepData` at all.

But perhaps this discussion isn't so important because `StepData` is supposed to be refactored anyhow? IMO my change is still worth it because it makes the function look more clear and helps get rid of the `type: ignore`. :)

Reply:
This part will probably change very Soon (TM), with `StepData` refactoring, so I wouldn't invest too much time into this. Accepting `StepData` is correct, returning them isn't, because `listify()` is never going to return a `StepData` instance. I would add `# type: ignore` comment, probably with more fitting error code, and a big `TODO: should go away with StepData becoming a dataclass` comment, and call it a day :)

Reply:
+1
@guy9050 pls make the change so we can merge this, it is blocking #1300

Reply:
and also rebase

Reply:
Done